stata_kernel/kernel.py

```diff
@@ -168,15 +170,22 @@ def do_is_complete(self, code):
     def do_complete(self, code, cursor_pos):
         """Provide context-aware suggestions
         """
-        env, pos, chunk, rcomp = self.completions.get_env(
-            code[:cursor_pos], code[cursor_pos:(cursor_pos + 2)],
-            self.sc_delimit_mode)
-
-        return {
-            'status': 'ok',
-            'cursor_start': pos,
-            'cursor_end': cursor_pos,
-            'matches': self.completions.get(chunk, env, rcomp)}
+        if self.completions.on:
+            env, pos, chunk, rcomp = self.completions.get_env(
+                code[:cursor_pos], code[cursor_pos:(cursor_pos + 2)],
+                self.sc_delimit_mode)
+
+            return {
+                'status': 'ok',
+                'cursor_start': pos,
+                'cursor_end': cursor_pos,
+                'matches': self.completions.get(chunk, env, rcomp)}
+        else:
+            return {
+                'status': 'ok',
+                'cursor_start': cursor_pos,
+                'cursor_end': cursor_pos,
+                'matches': []}
```

Comment:
So this is a setting for turning completions off entirely? At least in Hydrogen this can also be turned off client side. Does the kernel need a setting for it? In Jupyter notebook and the console you only see it when you use `tab`, and there's not really another use for `tab` in those environments...

Reply:
I can take it out. I think this was legacy from there I was concerned about it causing any lag. But even if the user has a ton of things defined I don't think the lag will be appreciable.

Reply:
Ah I see. If you find yourself bothered by lag we can put it back in.